### PR TITLE
Add wmt translation example

### PR DESCRIPTION
**Pull request by patrickvonplaten**
PR adds translation example for T5. 
It uses the `sacrebleu` BLEU scorer. 

I adapted the README.md a bit so that users are aware that models in official paper were attained with finetuned T5 @craffel

### CODE REVIEW

File: examples/translation/t5/evaluate_wmt.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        for hypothesis in dec:
+            fout.write(hypothesis + "\n")
+            fout.flush()
+
+
+def calculate_bleu_score(output_lns, refs_lns, score_path):
+    bleu = corpus_bleu(output_lns, [refs_lns])
+    result = "BLEU score: {}".format(bleu.score)
+    score_file = Path(score_path).open("w")
+    score_file.write(result)
```
Comment (LysandreJik) [inline examples/translation/t5/evaluate_wmt.py]: That's probably the simplest evaluation code I've seen, nice!

File: examples/translation/t5/evaluate_wmt.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        "output_path", type=str, help="where to save translation",
+    )
+    parser.add_argument(
+        "reference_path", type=str, help="like wmt/newstest2013.de",
+    )
+    parser.add_argument(
+        "score_path", type=str, help="where to save the bleu score",
+    )
+    parser.add_argument(
+        "--bs", type=int, default=16, required=False, help="batch size: how many to summarize at a time",
```
Comment (LysandreJik) [inline examples/translation/t5/evaluate_wmt.py]: Do you think this could be renamed to a more eloquent "batch_size" instead of "bs"?

**Review by LysandreJik — COMMENTED** [review]
Cool, that looks clean. I guess we don't need all the usual arguments (data parallel, fp16 and others) since this is an evaluation script and not a training script?

It doesn't seem to run on GPU, does it take a long time to compute? Shouldn't we try to cast to a GPU if there's one available?

File: examples/translation/t5/evaluate_wmt.py (hunk 2 above)
Comment (patrickvonplaten) [reply examples/translation/t5/evaluate_wmt.py]: yes!

**Comment by patrickvonplaten** [comment]
> long

Yeah we should definitely try to run it on a GPU - will take a look at that :-)

**Comment by patrickvonplaten** [comment]
Not sure whether we need fp16 and multi-gpu training. I think single GPU training is enough and t5 + wmt does not take much memory. But happy to take a look into it if you guys think it's worth it :-) @thomwolf @LysandreJik @julien-c

**Comment by patrickvonplaten** [comment]
Code quality test fails because of unpinned isort library (see https://github.com/huggingface/transformers/pull/3449)

**Review by thomwolf — APPROVED** [review]
Looks good to me

File: examples/translation/t5/README.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
+***This script evaluates the [T5 Model](https://arxiv.org/pdf/1910.10683.pdf) ``t5-base`` on the English to German WMT dataset. Please note that the results in the paper were attained using a ``t5-base`` model fine-tuned on translation, so that results will be slightly worse here***
```
Comment (craffel) [inline examples/translation/t5/README.md]: 1. Suggest changing "the [T5 Model](https://arxiv.org/pdf/1910.10683.pdf) ``t5-base``" to "the multitask pre-trained checkpoint for [``t5-base``](https://arxiv.org/pdf/1910.10683.pdf)
2. If you want to be more specific than "slightly", it looks like they are about 1.5 BLEU points lower than they would be with fine-tuning.

**Review by LysandreJik — APPROVED** [review]
LGTM!

**Pull request merged**